### PR TITLE
Maclaurin series approximation

### PR DESCRIPTION
**Pull request by antonAce**
Please include a summary of the change. Please also include relevant motivation and context (if applicable).

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have made corresponding changes to the comments
- [x] I have made corresponding changes to the README.md

Maclaurin series approximation is implemented for non-linear functions **exp(x)**, **cos(x)**, **sin(x)** in forms of **last term error series** and **set of terms**. This algorithm is an example how computer can calculate value of non-linear functions without using hardcoded argument-value tables.

### CODE REVIEW

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- Algorithms.Tests/Numeric/Decomposition/MaclaurinTests.cs  1
- Algorithms/Numeric/Series/Maclaurin.cs  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6091660&bid=19945759)

File: README.md [hunk 1]
```diff
@@ -21,6 +21,8 @@ This repository contains algorithms and data structures implemented in C# for ed
 		* [Naive solver](./Algorithms/Knapsack/NaiveKnapsackSolver.cs)
 		* [Dynamic Programming solver](./Algorithms/Knapsack/DynamicProgrammingKnapsackSolver.cs)
 	* [Numeric](./Algorithms/Numeric/)
+	    * [Series](./Algorithms/Numeric/Series/)
```
Comment (siriak) [inline README.md:24]: Codacy found an issue: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6091660)

File: README.md [hunk 2]
```diff
@@ -21,6 +21,8 @@ This repository contains algorithms and data structures implemented in C# for ed
...
+	        * [Maclaurin](./Algorithms/Numeric/Series/Maclaurin.cs)
```
Comment (siriak) [inline README.md:25]: Codacy found an issue: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 2 spaces](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6091660)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- Algorithms.Tests/Numeric/Decomposition/MaclaurinTests.cs  1
- Algorithms/Numeric/Series/Maclaurin.cs  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6091660&bid=19945759)

File: README.md [hunk 3]
```diff
@@ -28,6 +28,8 @@ This repository contains algorithms and data structures implemented in C# for ed
 			* [Binary GCD](./Algorithms/Numeric/GreatestCommonDivisor/BinaryGreatestCommonDivisorFinder.cs)
 		* [Factorization](./Algorithms/Numeric/Factorization)
 		    * [Trial division](./Algorithms/Numeric/Factorization/TrialDivisionFactorizer.cs)
+		* [Series](./Algorithms/Numeric/Series/)
+	    	* [Maclaurin](./Algorithms/Numeric/Series/Maclaurin.cs)
```
Comment (siriak) [inline README.md:32]: Codacy found an issue: [[list-item-content-indent] Don’t use mixed indentation for children, remove 1 space](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6091660)

Comment (siriak) [inline README.md:32]:
```suggestion
		* [Maclaurin](./Algorithms/Numeric/Series/Maclaurin.cs)
```
Thanks for the contribution, I'll review and merge this PR a bit later as I'm rather busy at the moment

File: README.md [hunk 4]
```diff
@@ -28,6 +28,8 @@ This repository contains algorithms and data structures implemented in C# for ed
 			* [Binary GCD](./Algorithms/Numeric/GreatestCommonDivisor/BinaryGreatestCommonDivisorFinder.cs)
 		* [Factorization](./Algorithms/Numeric/Factorization)
 		    * [Trial division](./Algorithms/Numeric/Factorization/TrialDivisionFactorizer.cs)
+		* [Series](./Algorithms/Numeric/Series/)
+		* [Maclaurin](./Algorithms/Numeric/Series/Maclaurin.cs)
```
Comment (siriak) [inline README.md:32]:
```suggestion
			* [Maclaurin](./Algorithms/Numeric/Series/Maclaurin.cs)
```

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Algorithms.Tests/Numeric/Decomposition/MaclaurinTests.cs  1
- Algorithms/Numeric/Series/Maclaurin.cs  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6091660&bid=19945759)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- Algorithms.Tests/Numeric/Decomposition/MaclaurinTests.cs  1
- Algorithms/Numeric/Series/Maclaurin.cs  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6091660&bid=19945759)

File: README.md [hunk 5]
```diff
@@ -28,6 +28,8 @@ This repository contains algorithms and data structures implemented in C# for ed
 			* [Binary GCD](./Algorithms/Numeric/GreatestCommonDivisor/BinaryGreatestCommonDivisorFinder.cs)
 		* [Factorization](./Algorithms/Numeric/Factorization)
 		    * [Trial division](./Algorithms/Numeric/Factorization/TrialDivisionFactorizer.cs)
+		* [Series](./Algorithms/Numeric/Series/)
+			* [Maclaurin](./Algorithms/Numeric/Series/Maclaurin.cs)
```
Comment (siriak) [inline README.md:32]: Codacy found an issue: [[list-item-content-indent] Don’t use mixed indentation for children, remove 1 space](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6091660)

Comment (antonAce) [reply README.md:32]: Both I and my IDE handles formatting pretty bad :sweat_smile:

Comment (siriak) [inline README.md:32]:
```suggestion
			* [Trial division](./Algorithms/Numeric/Factorization/TrialDivisionFactorizer.cs)
		* [Series](./Algorithms/Numeric/Series/)
			* [Maclaurin](./Algorithms/Numeric/Series/Maclaurin.cs)
```

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- Algorithms.Tests/Numeric/Decomposition/MaclaurinTests.cs  1
- Algorithms/Numeric/Series/Maclaurin.cs  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6091660&bid=19945759)

**Review by siriak — APPROVED** [review]
I'm waiting for #152 to be merged before this one to reflect real code coverage without those pesky intermittent changes in FastSearcher

**Comment by antonAce** [comment]
Okay, thank you for the review!)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- Algorithms.Tests/Numeric/Decomposition/MaclaurinTests.cs  1
- Algorithms/Numeric/Series/Maclaurin.cs  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6091660&bid=19945759)

**Review by siriak — APPROVED** [review]
Thanks!

**Pull request merged**